### DISCUSSION TITLE
I put in my API keys, I can only use Ollama but it doesn't understand Get-Clipboard or patterns.

### DISCUSSION THREAD

**Discussion by YorkyPoo** [danielmiessler/Fabric#291] [Q&A]
I think Mistral is trying to tell me how to fix it but I'm not sure what it means.
```
PS C:\Program Files\Code\fabric> fabric --model mistral:latest
Enter Question: Get-Clipboard | fabric --pattern extract_wisdom
 It looks like you're trying to use the PowerShell `Get-Clipboard` cmdlet to retrieve the contents of your clipboard, and then pass that data to the Fabric tool with a specific pattern called "extract\_wisdom". Unfortunately, there seems to be a misunderstanding here.

PowerShell's `Get-Clipboard` cmdlet is used to read the data currently stored on your system clipboard. It doesn't have any built-in support for piping that data directly to Fabric or any other external tool.

To use Fabric with PowerShell, you would typically save the clipboard contents as a variable in PowerShell and then pass it as an argument to Fabric. Here is an example:

```powershell
$clipboardContent = Get-Clipboard
Invoke-Expression ([System.Activator]::CreateInstance([Type]::GetType("Fabric.Client.FabricClient"))).UploadText("myAppName", "myKeyPath", $clipboardContent)
```

Replace `"myAppName"` and `"myKeyPath"` with the name of your Fabric application and the path to the key file, respectively. This PowerShell script saves the clipboard content as a variable, then uses Fabric's `UploadText` method to upload that text to your Fabric application.

As for the "extract\_wisdom" pattern you mentioned, it appears to be specific to Fabric and not something that is built into PowerShell or the Fabric CLI. You would need to consult the Fabric documentation to understand what that pattern does exactly.
```

**Comment by black-backdoor** [comment]
It seems like there might be a misunderstanding in how you're executing the command. Instead of submitting it as a question, you should execute it directly. Here's the corrected command:

```powershell
Get-Clipboard | fabric --model mistral:latest --pattern extract_wisdom
```

Allow me to explain this command:
- `Get-Clipboard` retrieves the contents of your clipboard in PowerShell.
- The `|` symbol pipes the output of `Get-Clipboard` to the Fabric tool.

**Reply by YorkyPoo** [reply]
Yeah, at the time I didn't know that. I still don't totally get it. When I was looking at the example there was not model. I tried selecting a default and then running 'pbpaste | fabric --pattern summarize' but It wasn't working. Then I was asking GPT-4 for awhile and I eventually found out to use  'Get-Clipboard' and 'mistral:latest'. I was out sick for 3 days trying to find that out and I wasn't able to find any resources or get answers. Some of that 3 days was trying to install it. The directions didn't work for me. Sort of like the problem I am having now. I came back because I wanted to play with the GUI. It looked like I just needed to update and run two commands. I am trying it in Virtual Studio because it was said to be easier. I am several hours into that but it keeps saying I'm missing stuff so I've been plugging it into GPT to help me install Node.js, python packages, poetry, libnss3, libatk1, electron, ffmeg, and update json. I am still getting npm errors on everything.

**Reply by black-backdoor** [reply]
> Yeah, at the time I didn't know that. I still don't totally get it. When I was looking at the example there was not model. I tried selecting a default and then running 'pbpaste | fabric --pattern summarize' but It wasn't working. Then I was asking GPT-4 for awhile and I eventually found out to use 'Get-Clipboard' and 'mistral:latest'. I was out sick for 3 days trying to find that out and I wasn't able to find any resources or get answers. Some of that 3 days was trying to install it. The directions didn't work for me. Sort of like the problem I am having now. I came back because I wanted to play with the GUI. It looked like I just needed to update and run two commands. I am trying it in Virtual Studio because it was said to be easier. I am several hours into that but it keeps saying I'm missing stuff so I've been plugging it into GPT to help me install Node.js, python packages, poetry, libnss3, libatk1, electron, ffmeg, and update json. I am still getting npm errors on everything.

I think the documentation needs some overhaul. I have already opened a discussion about this (#503)

**Reply by black-backdoor** [reply]
> Yeah, at the time I didn't know that. I still don't totally get it. When I was looking at the example there was not model. I tried selecting a default and then running 'pbpaste | fabric --pattern summarize' but It wasn't working. Then I was asking GPT-4 for awhile and I eventually found out to use 'Get-Clipboard' and 'mistral:latest'. I was out sick for 3 days trying to find that out and I wasn't able to find any resources or get answers. Some of that 3 days was trying to install it. The directions didn't work for me. Sort of like the problem I am having now. I came back because I wanted to play with the GUI. It looked like I just needed to update and run two commands. I am trying it in Virtual Studio because it was said to be easier. I am several hours into that but it keeps saying I'm missing stuff so I've been plugging it into GPT to help me install Node.js, python packages, poetry, libnss3, libatk1, electron, ffmeg, and update json. I am still getting npm errors on everything.

Did you follow the [quickstart](https://github.com/danielmiessler/fabric?tab=readme-ov-file#quickstart) guide?

About the pbpaste thing is a macOS only command.

**Reply by YorkyPoo** [reply]
When I first tried installing I used everything from quickstart to YT videos. I had, still have, no idea what commands are Mac/Linux/Windows. Some videos showed where to go and what is the best link to install Python. I messed with that for a while but later found out the one in the Windows store only worked for me at the time. I'm going to post on your discussion what I had to do.
